climpred/prediction.py

```diff
@@ -101,14 +162,37 @@ def compute_hindcast(
 
     Returns:
         skill (xarray object):
-            Predictability with main dimension ``lag``
+            Predictability with main dimension ``lag`` without dimension ``dim``
 
     """
+    if metric in PROBABILISTIC_METRICS:
+        if metric == 'e2r':
+            raise ValueError(
+                'Probabilistic metric `', metric, '` requires comparison `m2r`'
+            )
+        else:
+            stack = False
+    elif dim == 'init':
+        stack = True
+    elif dim == 'member':
+        stack = False
     nlags = max(hind.lead.values)
     comparison = get_comparison_function(comparison, HINDCAST_COMPARISONS)
+
+    forecast, reference = comparison(hind, reference, stack=stack)
```

Comment:
I know I've asked this in multiple places, so one answer somewhere will suffice. But why do we change to `stack` when the metric informs whether this is applied just over `init` or needs to take `member` into account?

Reply:
added a comment in the top of the compute functions